### ISSUE TITLE
Accept query params in initialRoute property

### ISSUE THREAD

**Issue by jwbaart** [testing-library/angular-testing-library#406]
We're using query params to configure our feature when loading and have a redirect in place when missing to set those. When using the initalRoute property, to prevent the redirect, query params aren't accepted([link](https://github.com/testing-library/angular-testing-library/blob/48629f71e9601211d9d8729810c87625c6e3d949/projects/testing-library/src/lib/testing-library.ts#L116)).
Navigation with query params is already available in a separate [function](https://github.com/testing-library/angular-testing-library/blob/48629f71e9601211d9d8729810c87625c6e3d949/projects/testing-library/src/lib/testing-library.ts#L169).

```
it('allows initially rendering a specific route with query parameters', async () => {
    const initialRoute = 'initial-route?param=query';
    const routes = [
      { path: initialRoute, component: FixtureComponent },
    ];
    await render(RouterFixtureComponent, {
      initialRoute,
      routes,
    });

    const router = TestBed.inject(Router);

    expect(router.url).toEqual(initialRoute);
    // could also be implemented by actually reading the params from ActivatedRoute but is more complex and the above does the same
  });
```

**Comment by jwbaart** [comment]
I'll see what I can manage :)

**Comment by jwbaart** [comment]
Prepared a more extensive test case to test new functionality: https://github.com/testing-library/angular-testing-library/compare/main...jwbaart:angular-testing-library:main
Now on to the actual work :)

**Comment by jwbaart** [comment]
Am I correct I couldn't find test watch functionality? I now monkey patched it with custom npm script(`"test:watch": "jest --watchAll"`) but expected it to already be present.

**Comment by timdeschryver** [comment]
> Am I correct I couldn't find test watch functionality? I now monkey patched it with custom npm script(`"test:watch": "jest --watchAll"`) but expected it to already be present.

😅 It seems like it yea.
You should be able to add a `--watch` flag to the existing test script, feel free to add it as `test:watch` if you want.

**Comment by jwbaart** [comment]
A bit later on I used `nx test testing-library --watchAll`, works even better :)

**Comment by timdeschryver** [comment]
Closed in #409
